### PR TITLE
Enabling NuGetPackageVerifier

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@Eilon

### CODE REVIEW

File: NuGetPackageVerifier.json [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            "AssemblyHasDocumentFileRule",
+            "AssemblyHasVersionAttributesRule",
+            "AssemblyHasServicingAttributeRule",
+            "SatellitePackageRule",
+            "StrictSemanticVersionValidationRule"
+        ],
+        "packages": {
+            "dnx-clr-win-x64": {
+                "VERSION_INFORMATIONALVERSION": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
```
Comment (victorhurdugaci) [inline NuGetPackageVerifier.json]: Why don't you use `/` instead of `\\`?

File: NuGetPackageVerifier.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                }
+            },
+            "dnx-mono": {
+                "VERSION_INFORMATIONALVERSION": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\lib\\Microsoft.Dnx.Tooling\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\Microsoft.Dnx.Host.Mono.dll": "Need to follow up, not from ADX team"
+                },
+                "VERSION_FILEVERSION": {
+                    "bin\\Microsoft.Dnx.Host.Mono.dll": "Need to follow up, not from ADX team"
```
Comment (victorhurdugaci) [inline NuGetPackageVerifier.json]: This is a file that we produce

File: NuGetPackageVerifier.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    "bin\\Microsoft.Framework.HashCodeCombiner.Sources.dll": "BUG: This file shouldn't be here",
+                    "bin\\mscorlib.dll": "Need to follow up, not from ADX team",
+                    "bin\\System.Security.SecureString.dll": "Need to follow up, not from ADX team"
+                }
+            },
+            "dnx-mono": {
+                "VERSION_INFORMATIONALVERSION": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\lib\\Microsoft.Dnx.Tooling\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\Microsoft.Dnx.Host.Mono.dll": "Need to follow up, not from ADX team"
```
Comment (victorhurdugaci) [inline NuGetPackageVerifier.json]: We produce this file

File: NuGetPackageVerifier.json (hunk 1 above)
Comment (ajaybhargavb) [inline NuGetPackageVerifier.json]: The exclusions are printed here https://github.com/aspnet/DnxTools/blob/dev/src/NuGetPackageVerifier/Program.cs#L230. It prints it out that way and it is easier to copy as is.

File: NuGetPackageVerifier.json (hunk 2 above)
Comment (victorhurdugaci) [inline NuGetPackageVerifier.json]: Here it is: https://github.com/aspnet/dnx/tree/dev/src/Microsoft.Dnx.Host.Mono

Comment (ajaybhargavb) [inline NuGetPackageVerifier.json]: :+1: There is and issue for this https://github.com/aspnet/dnx/issues/2045. Will update the  justification.

File: NuGetPackageVerifier.json (hunk 3 above)
Comment (ajaybhargavb) [inline NuGetPackageVerifier.json]: https://github.com/aspnet/dnx/pull/2717#discussion_r39696368

File: NuGetPackageVerifier.json [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        "packages": {
+            "dnx-clr-win-x64": {
+                "VERSION_INFORMATIONALVERSION": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\lib\\Microsoft.Dnx.Tooling\\Newtonsoft.Json.dll": "3rd party DLLs don't need this"
+                },
+                "SERVICING_ATTRIBUTE": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Microsoft.Framework.NotNullAttribute.Sources.dll": "BUG: This file shouldn't be here",
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\lib\\Microsoft.Dnx.Tooling\\Microsoft.Framework.NotNullAttribute.Sources.dll": "BUG: This file shouldn't be here",
```
Comment (Eilon) [inline NuGetPackageVerifier.json]: Are all the .Sources DLLs really still in the NUPKG? That sounds like a build bug that needs to be fixed ASAP...

Comment (muratg) [inline NuGetPackageVerifier.json]: I looked into a few .Sources packages, and this seems to be true.  I filed bugs on Common and Dnx repos to fix this:

https://github.com/aspnet/Common/issues/41
https://github.com/aspnet/dnx/issues/2720

Other repos also have some .Sources packages so we'll need to identify each and fix this.

**Comment by davidfowl** [comment]
I have lots of feedback for this pr related to hardcore the contents of packages. We should generate that. This will be brittle with the changes we are going to make to he DNX

File: NuGetPackageVerifier.json [hunk 5]
```diff
@@ -0,0 +1,134 @@
+{
+    "adx": { // Packages written by the ADX team and that ship on NuGet.org
```
Comment (troydai) [inline NuGetPackageVerifier.json]: Do we have to put comments in the Json?

Comment (Eilon) [inline NuGetPackageVerifier.json]: It's nice here :smile:

**Comment by ajaybhargavb** [comment]
Updated. @Eilon @pranavkm

File: NuGetPackageVerifier.json [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                "VERSION_INFORMATIONALVERSION": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\lib\\Microsoft.Dnx.Tooling\\Newtonsoft.Json.dll": "3rd party DLLs don't need this"
+                },
+                "SERVICING_ATTRIBUTE": {
+                    "bin\\lib\\Microsoft.Dnx.DesignTimeHost\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\lib\\Microsoft.Dnx.Tooling\\Newtonsoft.Json.dll": "3rd party DLLs don't need this",
+                    "bin\\Microsoft.CodeAnalysis.CSharp.dll": "Need to follow up, not from ADX team",
+                    "bin\\Microsoft.CodeAnalysis.dll": "Need to follow up, not from ADX team",
+                    "bin\\Microsoft.Dnx.Runtime.Internals.dll": "BUG: This file shouldn't be here"
```
Comment (Eilon) [inline NuGetPackageVerifier.json]: Is this tracked somewhere?

**Comment by Eilon** [comment]
Just had one question, so after that :shipit:

**Comment by BrennanConroy** [comment]
Travis failure?

**Comment by ajaybhargavb** [comment]
@BrennanConroy fixed. There was a problem with a comment in `makefile.shade`. Travis should build now.

**Comment by BrennanConroy** [comment]
Cool, talk to me before pushing.

File: NuGetPackageVerifier.json (hunk 6 above)
Comment (ajaybhargavb) [inline NuGetPackageVerifier.json]: Not sure. I couldn't find one. @BrennanConroy?

Comment (BrennanConroy) [inline NuGetPackageVerifier.json]: Just a comment in a PR I think https://github.com/aspnet/dnx/pull/2749#discussion_r39922395

Comment (Eilon) [inline NuGetPackageVerifier.json]: @BrennanConroy can you log a bug on this?

**Pull request merged**